src/db/repository.py

```
@@ -0,0 +1,767 @@
+        with self._get_session() as session:
+            stmt = select(Podcast).where(Podcast.feed_url == feed_url)
+            return session.scalar(stmt)
+
+    def list_podcasts(
+        self, subscribed_only: bool = True, limit: Optional[int] = None
+    ) -> List[Podcast]:
+        """List all podcasts, optionally filtering by subscription status."""
+        with self._get_session() as session:
+            stmt = select(Podcast)
+            if subscribed_only:
+                stmt = stmt.where(Podcast.is_subscribed == True)
```

Ruff (0.14.7)

Avoid equality comparisons to `True`; use `Podcast.is_subscribed:` for truth checks

Replace with `Podcast.is_subscribed`

(E712)